src/ssl_load.c

```diff
@@ -4146,6 +4146,84 @@ int wolfSSL_CTX_use_AltPrivateKey_Label(WOLFSSL_CTX* ctx, const char* label,
 #endif /* WOLFSSL_DUAL_ALG_CERTS */
 #endif /* WOLF_PRIVATE_KEY_ID */
 
+#if defined(WOLF_CRYPTO_CB) && !defined(NO_CERTS)
+
+static int wolfSSL_CTX_use_certificate_ex(WOLFSSL_CTX* ctx,
+    const char *label, const unsigned char *id, int idLen, int devId);
+
+static int wolfSSL_CTX_use_certificate_ex(WOLFSSL_CTX* ctx,
+    const char *label, const unsigned char *id, int idLen, int devId)
+{
+    int ret;
+    byte *certData = NULL;
+    word32 certDataLen = 0;
+    word32 labelLen = 0;
+
+    WOLFSSL_ENTER("wolfSSL_CTX_use_certificate_label_ex");
+
+    if (label != NULL) {
+        labelLen = (word32)XSTRLEN(label);
+    }
+
+    ret = wc_CryptoCb_GetCert(devId, (const sword8 *)label,
```

Comment:
Consider support for returning ASN1 or PEM...

Reply:
So PKCS11 spec says underlying data for CKA_VALUE for a CKC_X_509 object should be DER encoded, so that is what I assumed here as I have no way of knowing without attempting to check.

Reply:
That’s fine but the crypto callback needs to be designed more generically to support either. The result is just past to ProcessBuffer and it supports both ASN1 and PEM

Reply:
Ah I see now, yes the cryptocb itself should support an out variable to have the underlying cb implementation tell caller which it is in case future cb has pem underneath. I will make the changes.